packages/@aws-cdk/aws-cloudfront/lib/private/cache-behavior.ts

```diff
@@ -21,14 +20,10 @@ export interface CacheBehaviorProps extends BehaviorOptions {
  * CloudFrontWebDistribution implementation.
  */
 export class CacheBehavior {
+  private readonly originId: string;
 
-  /**
-   * Origin that this behavior will route traffic to.
-   */
-  public readonly origin: Origin;
-
-  constructor(private readonly props: CacheBehaviorProps) {
-    this.origin = props.origin;
+  constructor(originId: string, private readonly props: CacheBehaviorProps) {
+    this.originId = originId;
```

Comment:
Is this change due to potential conflicts with the ordering of calls to bind()?

This _is_ an internal class, but the interface feels pretty odd now:

```ts
  const behavior = new CacheBehavior('origin_id', {
    origin,
    pathPattern: '*',
  });
```

Seems a bit odd to pass in an Origin but not use it, and pass in the Origin id.

Maybe this suggests that this should take in a (possibly renamed) `AddCacheBehaviorProps` (without the Origin). It's a nitpick, but it would clean up the tests a bit.

Reply:
Yep, you're absolutely right! Changed.